core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/LDAPSearchSpecification.java

```diff
@@ -72,6 +77,16 @@ public LDAPSearchSpecification(RealmConfiguration realmConfig, List<ExpressionCo
         setLDAPSearchParamters(expressionConditions);
     }
 
+    public LDAPSearchSpecification(RealmConfiguration realmConfig, List<ExpressionCondition> expressionConditions,
```

Comment:
Following logic seems to be missed in this constructor which is available in the original constructor. In case this constructor used with `isGroupAttributeFiltering` false, following logic needed to be available in here too, right?

[1] 
```
        for (ExpressionCondition expressionCondition : expressionConditions) {
            if (ExpressionAttribute.ROLE.toString().equals(expressionCondition.getAttributeName())
                    && isGroupFiltering) {
                isMultiGroupFiltering = true;
            } else if (ExpressionAttribute.ROLE.toString().equals(expressionCondition.getAttributeName())) {
                isGroupFiltering = true;
            } else if (ExpressionAttribute.USERNAME.toString().equals(expressionCondition.getAttributeName())) {
                isUsernameFiltering = true;
            } else {
                isClaimFiltering = true;
            }
        }
```

Reply:
Fixed